### PR TITLE
Setup tests.

### PR DESCRIPTION
**Pull request by adilfulara**
**Summary**
- Add travis CI yml
- Add travis ci build badge
- Add coveralls integration for coverage badge
- Add nyc, mocha, chai and ignore-styles, enzyme, jsdom & react-addons-test-utils npm packages
- Add npm scripts cover and cover:report
- Add test/setup.js to setup dom for enzyme.mount()
- Add tests for number_widget.jsx

**Todo for upstream maintainer:** 
- Update README for urls to coveralls and travis ci to accurate badge numbers.

### CODE REVIEW

**Comment by adilfulara** [comment]
Badges are driven by 

https://coveralls.io/github/adilfulara/handsome
https://travis-ci.org/adilfulara/handsome

File: README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+[![Build Status](https://travis-ci.org/adilfulara/handsome.svg?branch=test-setup)](https://travis-ci.org/adilfulara/handsome)
+[![Coverage Status](https://coveralls.io/repos/github/adilfulara/handsome/badge.svg?branch=test-setup)](https://coveralls.io/github/adilfulara/handsome?branch=test-setup)
```
Comment (stephenyeargin) [inline README.md:2]: Would likely want to change the username on these to `davefp` before merging for practical reasons.

**Review by stephenyeargin — CHANGES_REQUESTED** [review]

**Comment by davefp** [comment]
Thanks @adilfulara, sorry this took so long for me to get round too

**Comment by adilfulara** [comment]
@davefp I forgot to do the changes suggested by @stephenyeargin.

**Comment by davefp** [comment]
I'll do it directly 👍

**Pull request merged**